### PR TITLE
[DRFT-697] Performance improvement of group comparison part of comparison report

### PR DESCRIPTION
**Pull request by petrblaho**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: drift/info_parser.py
Comment (jpramos123) [inline drift/info_parser.py:159]: @petrblaho  May we add a measurement here?

Comment (petrblaho) [reply drift/info_parser.py:159]: then we would be measuring the same as this https://github.com/RedHatInsights/drift-backend/pull/325/files/d2d06c01cf43768358e8f5b84a1b30d1f2b85a50#diff-d94d1e0636fff7ab5a4d4f6f24ff091770a8a2ec2c170453ec644766a4f16857R86

**Review by jpramos123 — APPROVED** [review]
LGTM! This is working with the pr_check and code looks consistent.

**Comment by sthirugn** [comment]
Thank you, Petr. I see Joao already approved, but let's wait until @msager27 to get this PR merged so we can do good testing coverage. Also this will help us to release what we have in master to Prod tomorrow.

**Comment by petrblaho** [comment]
@sthirugn I agree and wait with merging.

I will just squash some commits together so it has better git history and update it.

**Comment by petrblaho** [comment]
Squashed into two commits

**Pull request merged**
